server/api/v1/controllers/quotesApiController.js

```diff
@@ -0,0 +1,279 @@
+            message: "No quoteId provided"
+        });
+    }
+
+    quoteId = parseInt(quoteId);
+    if (isNaN(quoteId)) {
+        return res.status(400).send({
+            status: "error",
+            message: "Invalid quoteId provided"
+        });
+    }
+
+    let quote = await quotesManager.getQuote(quoteId);
+
+    if (quote == null) {
+        return res.status(404).send({
+            status: "error",
+            message: "Quote " + quoteId + " not found"
+        });
+    }
+
+    if (quotePatch.text != null && quotePatch.text !== "") {
+        quote.text = quotePatch.text;
+    }
+
+    if (quotePatch.originator != null && quotePatch.originator !== "") {
+        quote.originator = quotePatch.originator.replace(/@/g, "");
+    }
+    
+    if (quotePatch.creator != null && quotePatch.creator !== "") {
+        quote.creator = quotePatch.creator.replace(/@/g, "");
+    }
+
+    if (quotePatch.game != null && quotePatch.game !== "") {
+        quote.game = quotePatch.game;
+    }
+
+    if (quotePatch.createdAt && quotePatch.createdAt != null && quotePatch.createdAt !== "") {
+        quote.createdAt = moment(quotePatch.createdAt).toISOString();
+    }
+
+    try {
+        quote = await quotesManager.updateQuote(quote);
+    } catch (e) {
+        return res.status(500).send({
+           status: "error",
+           message: "Error updating quote " + quoteId + ": " + e
+        });
+    }
+
+    return res.json(formatQuote(quote));
+}
+
+exports.deleteQuote = async function(req, res) {
+    let { quoteId } = req.params;
+
+    if (quoteId == null) {
+        return res.status(400).send({
+            status: "error",
+            message: "No quoteId provided"
+        });
+    }
+
+    quoteId = parseInt(quoteId);
+    if (isNaN(quoteId)) {
+        return res.status(400).send({
+            status: "error",
+            message: "Invalid quoteId provided"
+        });
+    }
+
+    const quote = await quotesManager.getQuote(quoteId);
+
+    if (quote == null) {
+        return res.status(404).send({
+            status: "error",
+            message: "Quote " + quoteId + " not found"
+        });
+    }
+
+    await quotesManager.removeQuote(quoteId);
+
+    return res.status(204).send();
+};
+
+// Formats a quote for API output
+function formatQuote(quote) {
+    return {
+        id: quote._id,
+        text: quote.text,
+        originator: quote.originator,
+        creator: quote.creator,
+        game: quote.game,
+        createdAt: quote.createdAt
+    };
+}
+
+// Validates that all required quote fields are present and valid
+function validateQuote(quote) {
+    quote = quote ?? {};
+    let validationErrors = [];
+    
+    if (!quote.text || quote.text == null || quote.text === "") {
+        validationErrors.push("Missing quote text");
+    }
+    
+    if (!quote.creator || quote.creator == null || quote.creator === "") {
+        validationErrors.push("Missing quote creator");
+    }
```

Comment:
Another good opportunity for optional chaining here. `if(quote.text?.length < 1) { ... }`, etc

Reply:
Fixed. Negative checking is a little weirder with optional chaining, so it was either something like `if (!quote.text || quote.text?.length < 1)` or `if (!(quote.text?.length > 0)` so I opted for the latter to be concise.